sbtc-core/src/signer/mod.rs

```diff
@@ -57,6 +57,12 @@ pub trait Sign {
     fn verify_message(&self, public_key: &ecdsa::PublicKey, message: &[u8]) -> SBTCResult<bool>;
 }
 
+/// Revealer trait for revealing BTC commit transactions
+trait Revealer {
+    /// Create a reveal transaction from the BTC commit transaction
+    fn reveal_transaction(&self, tx: &BitcoinTransaction) -> SBTCResult<BitcoinTransaction>;
+}
+
```

Comment:
We should add at least a placeholder type here to represent the taproot internal key and the merkle tree of scripts, which is needed to create the reveal transaction. This type could potentially be used: https://docs.rs/bitcoin/0.30.1/bitcoin/taproot/struct.TaprootSpendInfo.html

Reply:
Yes let me give the doc a better read over so I can better reflect in the trait.

Reply:
the more I think about this, the more I don't think this belongs in the signer. Signer's should always be able to deny or approve a transaction and they should be governing the protocol itself. They should only be searching for the already revealed transaction. Not doing the logic to create it especially as this may mean that a signer has to be responsible for returning the funds to the initial committer( if the sBTC tx gets denied )which violates the idea that they may not want to interact with this entity. This should be an agreement between the committer and revealer, a one to one relationship, not a one to many essentially.

Reply:
Yeah we need to sync again. I agree with the point that I would like to have the Signer and Revealer be separate entities.

However, enforcing the signers to also be the Revealer does allow the signers to deny requests by simply ignoring commit transactions. Since commit transactions are invisible on-chain, no one knows that they are payments to the signers until they are revealed. If they are never revealed, the corresponding BTC will never be deposited. This is the mechanism we're piggy backing on to use commit-reveal to allow signers to deny transactions.

But won't ignoring commit transactions lock up the BTC forever? No, since the commit transaction may commit to multiple spend paths - the person creating the commit transaction can have a reclaim spend path allowing them to take back their BTC if the reveal tx never happens.

I'd love to chat more of this. I feel this is part of the design that we have taken for granted for too long, and there are probably better ways to do it than what's currently on the table.